repos/spack_repo/builtin/packages/py_heat/package.py

```diff
@@ -36,30 +39,85 @@ class PyHeat(PythonPackage):
         default=False,
         description="Use py-scikit-learn and py-matplotlib for the example tests",
     )
+    variant("cuda", default=False, description="build Py_Torch dependency with cuda support")
+    variant("rocm", default=False, description="build Py_Torch dependency with rocm support")
 
     depends_on("py-setuptools", type="build")
 
     with when("@1.3"):
         depends_on("python@3.8:3.10", type=("build", "run"))
         depends_on("py-mpi4py@3:", type=("build", "run"))
         depends_on("py-numpy@1.20:1", type=("build", "run"))
-        depends_on("py-torch@1.8:2.0.1", type=("build", "run"))
         depends_on("py-scipy@0.14:", type=("build", "run"))
         depends_on("pil@6:", type=("build", "run"))
         depends_on("py-torchvision@0.8:", type=("build", "run"))
 
+        with when("~cuda~rocm"):
```

Comment:
I think you can avoid this `when` context manager (the package seems to always depend on `py-torch@1.8:2.0.1`, it just gains more constraints when `+cuda` or `+rocm`)